src/main/kotlin/org/jmailen/gradle/kotlinter/support/EditorConfigUtils.kt

```diff
@@ -1,31 +1,75 @@
 package org.jmailen.gradle.kotlinter.support
 
-import com.pinterest.ktlint.core.KtLint
-import com.pinterest.ktlint.core.api.DefaultEditorConfigProperties
+import com.pinterest.ktlint.core.KtLintRuleEngine
 import com.pinterest.ktlint.core.api.EditorConfigOverride
+import com.pinterest.ktlint.core.api.editorconfig.EditorConfigProperty
+import org.ec4j.core.model.PropertyType
+import org.ec4j.core.model.PropertyType.PropertyValueParser.IDENTITY_VALUE_PARSER
 import org.gradle.api.file.ConfigurableFileCollection
 import org.gradle.api.file.ProjectLayout
 import org.gradle.api.logging.Logger
 import java.io.File
 
-internal fun editorConfigOverride(ktLintParams: KtLintParams): EditorConfigOverride {
-    val rules = ktLintParams.disabledRules
+internal fun editorConfigOverride(ktLintParams: KtLintParams) =
+    getPropertiesForDisabledRules(ktLintParams)
+        .plus(getPropertiesForExperimentalRules(ktLintParams))
+        .let(::buildEditorConfigOverride)
 
+private fun getPropertiesForDisabledRules(
+    ktLintParams: KtLintParams,
+): List<Pair<EditorConfigProperty<String>, String>> {
+    val rules = ktLintParams.disabledRules
     return if (rules.isEmpty()) {
-        EditorConfigOverride.emptyEditorConfigOverride
+        emptyList()
     } else {
-        EditorConfigOverride.from(DefaultEditorConfigProperties.ktlintDisabledRulesProperty to rules.joinToString(separator = ","))
+        rules
+            .asSequence()
+            .map(::getKtlintRulePropertyName)
+            .map { propertyName ->
+                EditorConfigProperty(
+                    type = PropertyType(propertyName, "Rule to be disabled", IDENTITY_VALUE_PARSER),
+                    defaultValue = "disabled",
+                )
+            }
+            .map { it to "disabled" }
+            .toList()
     }
 }
 
-internal fun resetEditorconfigCacheIfNeeded(
+private fun getPropertiesForExperimentalRules(ktLintParams: KtLintParams) =
+    if (ktLintParams.experimentalRules) {
+        listOf(
+            EditorConfigProperty(
+                type = PropertyType("ktlint_experimental", "Experimental rules", IDENTITY_VALUE_PARSER),
+                defaultValue = "enabled",
+            ) to "enabled",
+        )
+    } else {
+        emptyList()
+    }
```

Comment:
🔧 One last thing:
Since the goal of this PR is to support ktlint >=0.48, and they introduced a feature where you can hierarchically configure experimental rules via `.editorconfig` integration, I would strongly suggest to not block this behavior. With current state, if user enables experimental rules via plugin extension they won't be able to disable them via `.editorconfig` (i.e. for a specific directory). Having the "global" setting used as the default sounds much more reasonable to me. 

Keeping scope of this PR small - I'd suggest keeping the `editorConfigOverride` for disabled rules (to preserve backwards compatibility), but start using `editorConfigDefaults` for experimental rules (since that's a new thing we're adding right now).

Reply:
❔ 
And if the decision is to give plugin's extension priority, shouldn't we then _force_ the disabled state as well? To serve a _symmetrical_ behavior
```suggestion
private fun getPropertiesForExperimentalRules(ktLintParams: KtLintParams) =
	listOf(
	    EditorConfigProperty(
	        type = PropertyType("ktlint_experimental", "Experimental rules", IDENTITY_VALUE_PARSER),
	        defaultValue = "enabled",
	    ) to if(ktLintParams.experimentalRules) "enabled" to "disabled",
	)
```